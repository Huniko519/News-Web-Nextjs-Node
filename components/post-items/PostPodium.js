import React from 'react';
import PropTypes from 'prop-types';
import PostItemMedia from './PostItemMedia';
import PostItemBadge from './PostItemBadge';
import PostItemTitle from './PostItemTitle';
import PostItemLabel from './PostItemLabel';

/**
 *
 *
 * @param {post: {...postData}} props
 */
const PostHero = (props) => {
  const { post } = props;

  return (
    <div className="inews__post inews__post-podium show-author" key={post.id} data-post-id={post.id}>
      <PostItemBadge post={post} showAuthor />
      <div className="inews__post-podium__content">
        <PostItemMedia
          className="inews__post-podium__media"
          imageSize="small-1:1"
          post={post}
          height={155}
          width={155}
          layout="fixed"
          showAuthor
          placeholder="/static/images/placeholder/placeholder-84x84.png"
        />
        <div className="inews__post-podium__content-headline">
          <PostItemLabel post={post} />
          <PostItemTitle post={post} />
        </div>
      </div>
      <p className="inews__post-podium__excerpt" dangerouslySetInnerHTML={{ __html: post.excerpt.rendered }} />
    </div>
  );
};

/* istanbul ignore next */
PostHero.propTypes = {
  post: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default PostHero;
