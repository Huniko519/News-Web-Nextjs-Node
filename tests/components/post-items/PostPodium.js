import React from 'react';
import PostPodium from '../../../components/post-items/PostPodium';

test('renders the snapshot', () => {
  const post = {
    id: 1,
    title: {
      rendered: 'title',
    },
    excerpt: {
      rendered: 'excerpt',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };
  const postPodium = shallow(<PostPodium post={post} />);
  expect(postPodium).toMatchSnapshot();
});

it('should output with the correct class on the root element', () => {
  const post = {
    id: 1,
    title: {
      rendered: 'title',
    },
    excerpt: {
      rendered: 'excerpt',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };
  const postPodium = shallow(<PostPodium post={post} />);
  expect(postPodium.hasClass('inews__post')).toBe(true);
  expect(postPodium.hasClass('inews__post-podium')).toBe(true);
  expect(postPodium.hasClass('show-author')).toBe(true);
});

it('should call the <PostItemMedia /> component with the correct props', () => {
  const post = {
    id: 1,
    title: {
      rendered: 'title',
    },
    excerpt: {
      rendered: 'excerpt',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };

  const postPodium = shallow(<PostPodium post={post} />);
  const postItemMedia = postPodium.find('PostItemMedia');
  expect(postItemMedia.length).toBe(1);
  expect(postItemMedia.prop('className')).toBe('inews__post-podium__media');
  expect(postItemMedia.prop('imageSize')).toBe('small-1:1');
  expect(postItemMedia.prop('width')).toBe(155);
  expect(postItemMedia.prop('height')).toBe(155);
  expect(postItemMedia.prop('layout')).toBe('fixed');
  expect(postItemMedia.prop('placeholder')).toBe('/static/images/placeholder/placeholder-84x84.png');
  expect(postItemMedia.prop('showAuthor')).toBe(true);
  expect(postItemMedia.prop('post')).toMatchObject(post);
});

it('should call the <PostItemTitle /> component with the correct props', () => {
  const post = {
    id: 1,
    title: {
      rendered: 'title',
    },
    excerpt: {
      rendered: 'excerpt',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };

  const postPodium = shallow(<PostPodium post={post} />);
  const postItemMedia = postPodium.find('PostItemTitle');
  expect(postItemMedia.length).toBe(1);
  expect(postItemMedia.prop('post')).toMatchObject(post);
});

it('should output a <PostItemBadge /> component', () => {
  const post = {
    id: 1,
    title: {
      rendered: 'title',
    },
    excerpt: {
      rendered: 'excerpt',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };
  const postPodium = shallow(<PostPodium post={post} />);
  expect(postPodium.find('PostItemBadge').length).toBe(1);
  expect(postPodium.find('PostItemBadge').prop('post')).toMatchObject(post);
  expect(postPodium.find('PostItemBadge').prop('showAuthor')).toBe(true);
});

it('should output the post excerpt', () => {
  const post = {
    id: 1,
    title: {
      rendered: 'title',
    },
    excerpt: {
      rendered: 'excerpt',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };
  const postPodium = shallow(<PostPodium post={post} />);
  const excerptP = postPodium.find('p');
  expect(excerptP.length).toBe(1);
  expect(excerptP.hasClass('inews__post-podium__excerpt')).toBe(true);
  expect(excerptP.prop('dangerouslySetInnerHTML')).toMatchObject({
    __html: post.excerpt.rendered,
  });
});
