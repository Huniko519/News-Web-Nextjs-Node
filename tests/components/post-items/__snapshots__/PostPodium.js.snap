// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the snapshot 1`] = `
<div
  className="inews__post inews__post-podium show-author"
  data-post-id={1}
  key="1"
>
  <PostItemBadge
    post={
      Object {
        "breadcrumbs": Array [
          Object {
            "name": "News",
          },
        ],
        "excerpt": Object {
          "rendered": "excerpt",
        },
        "id": 1,
        "link": "https://inews-article.com/test-article",
        "title": Object {
          "rendered": "title",
        },
      }
    }
    showAuthor={true}
  />
  <div
    className="inews__post-podium__content"
  >
    <PostItemMedia
      className="inews__post-podium__media"
      height={155}
      imageSize="small-1:1"
      layout="fixed"
      linkToAuthor={false}
      placeholder="/static/images/placeholder/placeholder-84x84.png"
      post={
        Object {
          "breadcrumbs": Array [
            Object {
              "name": "News",
            },
          ],
          "excerpt": Object {
            "rendered": "excerpt",
          },
          "id": 1,
          "link": "https://inews-article.com/test-article",
          "title": Object {
            "rendered": "title",
          },
        }
      }
      postItemType=""
      showAuthor={true}
      width={155}
    />
    <div
      className="inews__post-podium__content-headline"
    >
      <PostItemLabel
        post={
          Object {
            "breadcrumbs": Array [
              Object {
                "name": "News",
              },
            ],
            "excerpt": Object {
              "rendered": "excerpt",
            },
            "id": 1,
            "link": "https://inews-article.com/test-article",
            "title": Object {
              "rendered": "title",
            },
          }
        }
      />
      <PostItemTitle
        post={
          Object {
            "breadcrumbs": Array [
              Object {
                "name": "News",
              },
            ],
            "excerpt": Object {
              "rendered": "excerpt",
            },
            "id": 1,
            "link": "https://inews-article.com/test-article",
            "title": Object {
              "rendered": "title",
            },
          }
        }
      />
    </div>
  </div>
  <p
    className="inews__post-podium__excerpt"
    dangerouslySetInnerHTML={
      Object {
        "__html": "excerpt",
      }
    }
  />
</div>
`;
